### ISSUE TITLE
SumProduct function in Excel for the web returns 0, or #CALC

### ISSUE THREAD

**Issue by EdwinStraver** [OfficeDev/office-js#3286]
I create a custom function, that returns a number between, say, 1 and 10. I put this custom function in cells A1, B1 and C1 in Excel for the web.
I place 10,20, 30 in cells A2,B2 and C2.
I compute =sumproduct(A1:C1, A2:C2) in cell A4.

I expect the sumproduct to be returned, but I keep getting a 0 (most often), or a #CALC (once in a while).

This only seems to be a problem with the sumproduct function. When I use SUM, SUMSQ, etc. it all computes fine.

Note that this only happens in Excel for the web, and not in Excel desktop. I have also tried it with several different custom functions, and sat the same problem.

**Comment by rundongmsft** [comment]
Hi @EdwinStraver, I couldn't reproduce this issue. Maybe the custom function we used is different, what is the custom function you created in your side? Can it be provided to me?

**Comment by EdwinStraver** [comment]
It is not so easy for me to share the custom function, but since the add-in is in the Microsoft Store, you can simply install the "Analytic Solver Addin", and then use the custom function =PsiNormal(1,2) in cell A1, B1, C1 in the scenario above. (PsiNormal will return a random number from a distribution with mean 1, and standard deviation 2.)

**Comment by rundongmsft** [comment]
Hi @EdwinStraver, I reproduced this issue and found it was due to the build-in function "sumproduct" in Excel online not handling volatile custom function result as parameter correctly, we are in contact with the owner team of "sumproduct". If this issue is urgent for you, please replace "sumproduct" with another build-in function as a workaround. For example, in your scenario, "=sumproduct(A1:C1, A2:C2) " can be replaced to "=sum(A1:C1*A2:C2)".

**Comment by EdwinStraver** [comment]
Thanks for the update!

**Comment by JinghuiMS** [comment]
Closing this issue because it appears to be resolved, but please let us know if you have further questions or feel it needs to be re-opened.

**Comment by EdwinStraver** [comment]
It is not fixed of the moment. Does the fix still need to be rolled out?

**Comment by EdwinStraver** [comment]
@JinghuiMS , @rundongmsft this issue is not fixed. Can you reopen ?

**Comment by JinghuiMS** [comment]
Does the workaround("=sumproduct(A1:C1, A2:C2) " can be replaced to "=sum(A1:C1*A2:C2)".) work for you?

**Comment by EdwinStraver** [comment]
Yes, the workaround works, but I can hardly tell the users of our add-in "Make sure you don't use the SUMPRODUCT" function...

**Comment by EdwinStraver** [comment]
It has been almost a year now; and the bug is still present. Is there any movement on this front?

**Comment by rundongmsft** [comment]
Hi @EdwinStraver, we created an internal bug item 8830648 to track this issue. We will let you know for any updates.

**Comment by EdwinStraver** [comment]
@rundongmsft  It has now been almost 3 years since this bug was reported.   
And its still an issue.....
Any progress at all?

**Comment by EdwinStraver** [comment]
It is now March, 2026.   This bug still hasn't been fixed :(

**Comment by deepanshugupta017** [comment]
@EdwinStraver 
Is this issue still occurring for you? I tried to reproduce it with Script lab custom function and analytic solver custom function, but it is non-reproducible for me. Attaching the repro video for you. Let me know if the issue still persists for you.

https://github.com/user-attachments/assets/5b02390e-84fe-43a0-abfe-6a2889e1500e

**Comment by EdwinStraver** [comment]
Thx for taking a look at this., but unfortunately, it still doesnt seem resolved. I attached a little video as well.

Edwin

https://github.com/user-attachments/assets/3e3acbb5-895a-4f56-b137-87fb2e6dccb2

**Comment by mw-msft** [comment]
Hi @EdwinStraver , thank you for your patience and for continuing to share feedback on this issue. We understand the impact this behavior has had and appreciate the time you've taken to provide details and help us investigate. After working through the investigation, we've confirmed that the required changes fall under another product team's ownership. We've shared our findings with the owning team and are transferring the issue, along with the customer impact you've highlighted, so it can continue to be tracked and reviewed by the appropriate owners. We sincerely appreciate your continued engagement and advocacy for this scenario.

**Comment by mw-msft** [comment]
Internal WI #11944686

**Comment by EdwinStraver** [comment]
I love how you guys send a notification right before a holiday weekend, that this titems will be closed in 3 days if no feedback is given when the bug has NOT been fixed in 5 years !!!!!!  And it still not fixed.
